tests/unit/ui/components/TestShowMoreCard.js

```diff
@@ -1,105 +1,223 @@
+            __html: '<span>This is different Text.</span>',
+          }}
+        />
+      ),
+    });
+    /* eslint-enable react/no-danger */
 
-    expect(root.childAt(0)).toHaveText('Hello I am description');
+    sinon.assert.called(dispatchSpy);
   });
 
-  it('executes truncateToMaxHeight when it recieves props', () => {
-    const root = mount(<ShowMoreCardBase i18n={fakeI18n()} />);
-    const component = root.instance();
+  it('dispatches if the children has changed', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
 
-    const contentNode = component.contents;
-    const truncateToMaxHeight = sinon.spy(component, 'truncateToMaxHeight');
-    root.setProps(); // simulate any kind of update to properties
+    const root = render({
+      store,
+      children: 'Some text',
+    });
 
-    sinon.assert.calledWith(truncateToMaxHeight, contentNode);
+    root.setProps({
+      children: 'Some new text',
+    });
+
+    sinon.assert.called(dispatchSpy);
+  });
+
+  it('does not dispatch if the html children is the same', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    /* eslint-disable react/no-danger */
+    const root = render({
+      store,
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>Some component text.</span>',
+          }}
+        />
+      ),
+    });
+
+    root.setProps({
+      children: (
+        <div
+          dangerouslySetInnerHTML={{
+            __html: '<span>Some component text.</span>',
+          }}
+        />
+      ),
+    });
+    /* eslint-enable react/no-danger */
+
+    sinon.assert.notCalled(dispatchSpy);
   });
 
-  it('expands if new child content is smaller', () => {
-    const root = mount(
-      <ShowMoreCardBase i18n={fakeI18n()}>
-        This would be very long content, but we cannot get `clientHeight` in the
-        tests, so this will be forced below (via setState()).
-      </ShowMoreCardBase>,
+  it('does not dispatch if the children is the same', () => {
+    const { store } = dispatchClientMetadata();
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = render({
+      store,
+      children: 'Some text',
+    });
+
+    root.setProps({
+      children: 'Some text',
+    });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('executes truncateToMaxHeight when it receives props', () => {
+    const root = render();
+
+    const contentNode = root.instance().contents;
+
+    const truncateToMaxHeight = sinon.spy(
+      root.instance(),
+      'truncateToMaxHeight',
     );
 
-    // We have to manually set the expanded flag to false because we don't have
-    // a `clientHeight` in the tests.
-    root.setState({ expanded: false });
+    // We are simulating the truncate method call.
+    root.instance().truncateToMaxHeight({ clientHeight: MAX_HEIGHT + 1 });
```

Comment:
Why do we have to call this first?

Reply:
I guess we don't need to. I was simulating the whole process (does this when it firsts renders too).  I'll remove this..